zerver/lib/types.py

```diff
@@ -221,6 +221,49 @@ class NeverSubscribedStreamDict(TypedDict):
     subscribers: NotRequired[List[int]]
 
 
+class StreamClientDict(TypedDict):
+    """Stream information provided to Zulip clients as a dictionary via API.
+    It should contain all the fields specified in `zerver.models.Stream.API_FIELDS`
+    with few exceptions and possible additional fields.
+    """
+
+    date_created: int
+    description: str
+    first_message_id: Optional[int]
+    history_public_to_subscribers: bool
+    stream_id: int  # `stream_id`` represents `id` of the `Stream` object in `API_FIELDS`
+    invite_only: bool
+    is_web_public: bool
+    message_retention_days: Optional[int]
+    name: str
+    rendered_description: str
+    stream_post_policy: int
+    # Additional fields not specified in `API_FIELDS`
+    is_announcement_only: bool
+    is_default: NotRequired[bool]
+
+
+class SubscriptionDict(StreamClientDict):
```

Comment:
Maybe `APISubscriptionDict`, if this is the format for an object sent to clients via the API?